src/Parsing/ParserState.php

```diff
@@ -328,7 +328,7 @@ public function isEnd(): bool
     }
 
     /**
-     * @param list<string>|string $stopCharacters
+     * @param list<string|null>|string|null $stopCharacters
```

Comment:
We can make the reasoning behind this a bit more obvious to the reader:

```suggestion
     * @param list<string|self::EOF>|string|self::EOF $stopCharacters
```